hail/python/hail/linalg/blockmatrix.py

```diff
@@ -2560,7 +2558,7 @@ def _shape_after_broadcast(left, right):
     """
 
     def join_dim(l_size, r_size):
-        if not (l_size == r_size or l_size == 1 or r_size == 1):
+        if l_size not in {r_size, 1}:
```

Comment:
oooo yep, that does seem wrong, this one is definitely only run under `--unsafe-fixes`, and i changed it to a `set` because it suggests doing that when the types are hashable, but i didn't catch that it had dropped the `rsize == 1` from the condition